src/coverage_reporter/api.cr

```diff
@@ -42,13 +42,27 @@ module CoverageReporter
       end
     end
 
-    def tls_for(uri : URI) : OpenSSL::SSL::Context::Client?
+    def tls_for(uri : URI, force_insecure_requests : Bool = false) : OpenSSL::SSL::Context::Client?
+      return OpenSSL::SSL::Context::Client.insecure if force_insecure_requests
       return nil unless uri.scheme == "https"
       return nil if uri.host == "coveralls.io"
 
       OpenSSL::SSL::Context::Client.insecure
     end
 
+    def with_ssl_errors_handling() : HTTP::Client::Response
+      begin
+        yield
+      rescue ex : OpenSSL::SSL::Error
+        if OpenSSLVersion.new.can_fail?
+          Log.error <<-ERROR
+            Consider upgrading `openssl` library to version >= #{OpenSSLVersion::WORKS} or using --force-insecure-requests flag
```

Comment:
Since we're distributing statically compiled version of the reporter for all Linux systems, upgrading `openssl` on the target machine wouldn't actually do anything. The issue here is incompatibility between the OpenSSL 3.0.9 included with the reporter, and the PKI infrastructure on the server (shared trusted root CA certs etc)